clpy/core/carray.pxi

```diff
@@ -175,7 +175,7 @@ cpdef function.Module compile_with_cache(
             proc.kill()
             source, errstream = proc.communicate()
 
-        if proc.returncode != 0 and len(errstream) > 0:
+        if proc.returncode != 0 or len(errstream) > 0:
```

Comment:
Can you explain details why we need this?
https://github.com/fixstars/clpy/issues/213#issuecomment-538908642

Some strings would be output in errstream?

Reply:
When ultima completed her work and she detected some error, `proc.returncode != 0 and len(errstream) > 0` equals to `True`.
However, when ultima met her demise like being hit `SIGSEGV` , `proc.returncode != 0` equals to `True` but `len(errstream) > 0` may not.
Current ClPy can't detect these pattern of ultima's failed, then our investigation for some problems of ultima has been complicated.
So it should be fixed I think.

[Ultima runs with `-w -Wno-narrowing`](https://github.com/fixstars/clpy/blob/cc72c544a17c823ab9af443366f2191b00746ba1/ultima/ultima.cpp#L3402-L3403) , so IMO, `proc.returncode == 0 and len(errstream) > 0` equals to `False` in normal ClPy use-cases.
If you wonder the situation that ultima reports some warnings, we can fix this code like below:

```
if proc.returncode != 0:
    if len(errstream) == 0:
        errstream = '(stderr was empty)'
    raise clpy.backend.ultima.exceptions.UltimaRuntimeError(
        proc.returncode, errstream)
```

Reply:
I understand. Some error (e.g. SIGSEGV) can be output no errstream.